backend/src/nodes/nodes/ncnn/upscale_image.py

```diff
@@ -69,16 +70,26 @@ def __init__(self):
     def upscale(
         self,
         img: np.ndarray,
-        net,
+        model: NcnnModel,
         input_name: str,
         output_name: str,
-        max_tile_size: Union[int, None],
+        tile_size: Union[int, None],
     ):
         exec_options = get_execution_options()
+        net = get_ncnn_net(model, exec_options)
         # Try/except block to catch errors
         try:
             vkdev = ncnn.get_gpu_device(exec_options.ncnn_gpu_index)
-            # logger.info(vkdev.get_heap_budget())
+            heap_budget = vkdev.get_heap_budget() * 1024 * 1024
+
+            max_tile_size = (
+                tile_size
+                if tile_size is not None
+                else estimate_tile_size(
+                    heap_budget, heap_budget, model.bin_length, img, 4, 0.8
```

Comment:
[Are you sure we need the 80% usage?](https://github.com/Tencent/ncnn/blob/bbbe17c5b502d4128968cf755ab533265289d620/src/gpu.cpp#L2907)

Reply:
That's only if `(!info.support_VK_EXT_memory_budget())`

Reply:
Good point.